src/pip/_internal/utils/misc.py

```diff
@@ -419,7 +425,7 @@ def user_test(d):
             return True
 
     # because of pkg_resources vendoring, mypy cannot find stub in typeshed
-    return [d for d in pkg_resources.working_set  # type: ignore
+    return [d for d in pkg_resources.WorkingSet(paths)  # type: ignore
```

Comment:
This will introduce incompatibilities when `paths` is `None`. `working_set` (which is built with `WorkingSet._build_master()` performs some extra checks when `__main__.__requires__` is set (a Setuptools implementation detail).

Reply:
Can you elaborate more on what the checks do exactly, and whether they're important for `pip freeze`? Also, should the same or similar checks be done in the https://github.com/pypa/pip/pull/6450 case?

Reply:
I am not 100% sure (setuptools is quite complicated), but AFAICT this is a mechanism to distinguish distributions when you have multiple dists in `sys.path`. Entry points installed by setuptools have a `__requires__` attribute that records where the script is from, so if you (say) have pip 18.0 and 19.1 in `sys.path`, a `__requires__ = ["pip==19.1"]` flag would help `pkg_resources` to correctly locate pip 19.1 (instead of 18.0) in the environment.

Reply:
@uranusjr thanks for the heads up. I'm not sure it makes sense to spend much more time on investigating how `setuptools` works for this small refactor, so unless someone with more `setuptools` knowledge can chime in whether this is OK, I'm happy to close the PR.

Reply:
I think this simplification was worth exploring (thank you for pursuing it, @lkollar), but I agree. If there's a risk of breakage here, I don't think it's worth it unless someone who knows can confirm it's okay.